### PR TITLE
SOE-646: add link block

### PR DESCRIPTION
**Pull request by cjwest**
Adding a block to that will link to the help page

### CODE REVIEW

**Comment by boznik** [comment]
this function is already getting used, it needs it's own name or moved into a utility module and referenced everywhere it is getting used:

array_merge_recursive_ex

File: stanford_help.install [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Install tasks.
+ */
+
+/**
+ * Implements hook_enable().
+ */
+function stanford_help_enable() {
```
Comment (sherakama) [inline stanford_help.install:16]: Can you add a little more to this function description to what it is doing and why?

File: stanford_help.install [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  // Set the variable for Content Access.
+  $current = variable_get('content_access_stanford_help', array());
+  $caspp = array('view_own' => $rids, 'view' => $rids);
+  $merged = array_merge_recursive_ex($current, $caspp);
+  variable_set('content_access_stanford_help', $merged);
+
+  node_access_rebuild();
+}
+
+function _stanford_help_array_merge_recursive_ex(array & $array1, array & $array2) {
```
Comment (sherakama) [inline stanford_help.install:58]: Missing doc block.

File: stanford_help.install [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  $merged = $array1;
+  foreach ($array2 as $key => & $value) {
+    if (is_array($value) && isset($merged[$key]) && is_array($merged[$key])) {
+      $merged[$key] = _stanford_help_array_merge_recursive_ex($merged[$key], $value);
+    } elseif (is_numeric($key)) {
```
Comment (sherakama) [inline stanford_help.install]: The elseif statement should be on the next line.

```
}
elseif () {
```

File: stanford_help.install [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  $caspp = array('view_own' => $rids, 'view' => $rids);
+  $merged = array_merge_recursive_ex($current, $caspp);
+  variable_set('content_access_stanford_help', $merged);
+
+  node_access_rebuild();
+}
+
+function _stanford_help_array_merge_recursive_ex(array & $array1, array & $array2) {
+  $merged = $array1;
+  foreach ($array2 as $key => & $value) {
```
Comment (sherakama) [inline stanford_help.install]: No space between & and $

File: stanford_help.module [hunk 5]
```diff
@@ -5,3 +5,123 @@
...
+
+      variable_set('stanford_help_settings', $new_settings);
+      break;
+  }
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function stanford_help_block_view($delta='') {
```
Comment (sherakama) [inline stanford_help.module]: spaces around the =

File: stanford_help.module [hunk 6]
```diff
@@ -5,3 +5,123 @@
...
+    case 'stanford_site_specific_help' :
+      $block['content'] = stanford_site_specific_help_block_view();
+      break;
+  }
+
+  return $block;
+}
+
+function stanford_site_specific_help_block_view()
+{
```
Comment (sherakama) [inline stanford_help.module]: This should be on the same line as the function statement.

**Comment by cjwest** [comment]
@sherakama - Thanks for your comments!
@boznik - It's ready for your review!

File: stanford_help.install [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  }
+  if ($admin_role = user_role_load_by_name('administrator')) {
+    $rids[] = $admin_role->rid;
+  }
+
+  // Set the variable for Content Access.
+  $current = variable_get('content_access_stanford_help', array());
+  $caspp = array('view_own' => $rids, 'view' => $rids);
+  $merged = array_merge_recursive_ex($current, $caspp);
```
Comment (boznik) [inline stanford_help.install]: this should be:
$merged = _stanford_help_array_merge_recursive_ex($current, $caspp);

**Pull request merged**